### PR TITLE
Deprecate DropdownMenu v1 + Remove DropdownMenu v2

### PR DESCRIPTION
**Pull request by siddharthkp**
- Built on top of https://github.com/primer/react/pull/1893 and https://github.com/primer/react/pull/1897, merge after them
- If we revert [^ #1893](https://github.com/primer/react/pull/1893) or [^ #1897](https://github.com/primer/react/pull/1897), we should also revert this PR
- Part of https://github.com/github/primer/issues/724

Deprecate Dropdown v1
- [x] Move from src/index to src/deprecated
- [x] Move to deprecated folder in docs/
- [x] Add 'deprecated' to doc examples 
- [x] Move to deprecated section in docs/nav.yml
- [x] Update status to deprecated in docs
- [x] Update import path in related components
- [x] Update title in story meta "Deprecated components"
- [x] Move to stories/deprecated/
- [x] Update import path in stories
- [x] Move to tests/deprecated/
- [x] Update import path in tests
- [x] Update test snapshots
- [x] Add deprecation notice in docs
- [x] Add @deprecated JSDoc annotation

Remove DropdownMenu v2
- [x] Remove from src/deprecated
- [x] Remove from docs/drafts/
- [x] Remove from docs/nav.yml
- [x] Remove stories
- [x] Remove tests
- [x] Update snapshots

Checks
- [x] Tests passing
- [x] Docs working
- [x] Storybook working

### CODE REVIEW

File: .changeset/deprecate-dropdownmenu.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+---
+'@primer/react': major
+---
+
+### DropdownMenu
+
+DropdownMenu has been deprecated in favor of ActionMenu with ActionList
```
Comment (rezrah) [inline .changeset/deprecate-dropdownmenu.md:7]: Thinking out loud. I don't think we need to advertise ActionList in this way as that's implementation detail.. or is this one of those `Built with ActionList™ 💪` kind of things 😄

File: docs/content/deprecated/DropdownMenu.mdx [hunk 2]
```diff
@@ -1,14 +1,91 @@
...
+}
+```
+
+Or continue using deprecated API:
+
+```js
+import {DropdownMenu} from '@primer/react/deprecated'
+```
+
+## Example - this is not working, need to debug why.
```
Comment (rezrah) [inline docs/content/deprecated/DropdownMenu.mdx:86]: Can we remove this before merge? 😅 
Screenshot 2022-02-25 at 20 43 27

File: docs/src/@primer/gatsby-theme-doctocat/components/live-preview-wrapper.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
-import {BaseStyles, Box, ThemeProvider, useTheme, DropdownMenu, DropdownButton} from '@primer/react'
+import {BaseStyles, Box, ThemeProvider, useTheme} from '@primer/react'
+import {DropdownMenu, DropdownButton} from '@primer/react/deprecated'
```
Comment (rezrah) [inline docs/src/@primer/gatsby-theme-doctocat/components/live-preview-wrapper.js:2]: 👍. Would be nice to mop this up next week and refactor to ActionMenu+ActionList.

Comment (rezrah) [reply docs/src/@primer/gatsby-theme-doctocat/components/live-preview-wrapper.js:2]: Would there be anything stopping us from doing this?

**Review by rezrah — APPROVED** [review]
LGTM

File: docs/src/@primer/gatsby-theme-doctocat/components/live-preview-wrapper.js (hunk 3 above)
Comment (siddharthkp) [reply docs/src/@primer/gatsby-theme-doctocat/components/live-preview-wrapper.js:2]: Nope, wanted to leave this here for next week :)

Comment (rezrah) [reply docs/src/@primer/gatsby-theme-doctocat/components/live-preview-wrapper.js:2]: 👍 , let's add this to the v35 checklist then

**Pull request merged**